### PR TITLE
Updated the github client

### PR DESCRIPTION
**Pull request by hpandeycodeit**
Updated the github client as per Issue #11478

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@hpandeycodeit thanks for the cleanup! I think this will be a larger change than just an import re-name, unfortunately. The two clients are similar but not identical.

**Comment by hpandeycodeit** [comment]
@stevekuznetsov Thank you for your input. I have updated the code as per the new library. Hopefully, it will be successful this time. If not, I will rework on it.

File: prow/prstatus/prstatus.go [hunk 1]
```diff
@@ -25,13 +25,12 @@ import (
 	"strings"
 	"time"
 
-	gogithub "github.com/google/go-github/github"
 	"github.com/gorilla/sessions"
 	githubql "github.com/shurcooL/githubv4"
 	"github.com/sirupsen/logrus"
 	"golang.org/x/oauth2"
+	gogithub "k8s.io/test-infra/prow/github"
```
Comment (stevekuznetsov) [inline prow/prstatus/prstatus.go]: nit: `test-infra` imports should be grouped together
non-nit: we should not call this `gogithub`

File: prow/prstatus/prstatus.go [hunk 2]
```diff
@@ -341,8 +340,15 @@ func (da *DashboardAgent) GetHeadContexts(ghc githubClient, pr PullRequest) ([]C
 }
 
 // GetUser attempts to get the currently authenticated Github user.
-func (da *DashboardAgent) GetUser(client *ghclient.Client) (*gogithub.User, error) {
-	return client.GetUser("")
+func (da *DashboardAgent) GetUser(client *gogithub.Client) (*gogithub.User, error) {
```
Comment (stevekuznetsov) [inline prow/prstatus/prstatus.go]: We should just change the code to use the BotName() function directly

Comment (midnightconman) [reply prow/prstatus/prstatus.go]: This should do the trick, did you come up with something similar @hpandeycodeit? 

https://github.com/kubernetes/test-infra/compare/master...midnightconman:prstatus-new-githubclient-20190225141936?expand=1

**Comment by midnightconman** [comment]
This change is related to issue https://github.com/kubernetes/test-infra/issues/11224

File: prow/prstatus/prstatus.go (hunk 2 above)
Comment (hpandeycodeit) [reply prow/prstatus/prstatus.go]: yes I am in process of making a commit :)

File: prow/prstatus/prstatus.go [hunk 3]
```diff
@@ -201,11 +199,13 @@ func (da *DashboardAgent) HandlePrStatus(queryHandler PullRequestQueryHandler) h
 		// chance to validate whether the access token is consumable or not. If
 		// not, we invalidate the sessions and continue as if not logged in.
 		token, ok := session.Values[tokenKey].(*oauth2.Token)
-		var user *gogithub.User
+		var user *github.User
+		var botName string
 		if ok && token.Valid() {
-			goGithubClient := ghclient.NewClient(token.AccessToken, false)
+			goGithubClient := github.NewClient(func() []byte { return []byte(token.AccessToken) }, githubEndpoint)
```
Comment (midnightconman) [inline prow/prstatus/prstatus.go]: Can we rename this to `githubClient`?

File: prow/prstatus/prstatus.go [hunk 4]
```diff
@@ -201,11 +199,13 @@ func (da *DashboardAgent) HandlePrStatus(queryHandler PullRequestQueryHandler) h
...
 			var err error
-			user, err = queryHandler.GetUser(goGithubClient)
+			botName, err = goGithubClient.BotName()
```
Comment (midnightconman) [inline prow/prstatus/prstatus.go]: Same here, `githubClient` instead of `goGithubClient`.

File: prow/prstatus/prstatus.go [hunk 5]
```diff
@@ -341,9 +341,16 @@ func (da *DashboardAgent) GetHeadContexts(ghc githubClient, pr PullRequest) ([]C
 }
 
 // GetUser attempts to get the currently authenticated Github user.
-func (da *DashboardAgent) GetUser(client *ghclient.Client) (*gogithub.User, error) {
-	return client.GetUser("")
-}
+//func (da *DashboardAgent) BotName(client *github.Client) (*github.User, error) {
```
Comment (midnightconman) [inline prow/prstatus/prstatus.go]: Can we delete this comment.

File: prow/prstatus/prstatus.go [hunk 6]
```diff
@@ -53,7 +51,7 @@ type githubClient interface {
 type PullRequestQueryHandler interface {
 	QueryPullRequests(context.Context, githubClient, string) ([]PullRequest, error)
 	GetHeadContexts(ghc githubClient, pr PullRequest) ([]Context, error)
-	GetUser(*ghclient.Client) (*gogithub.User, error)
+	// BotName(*github.Client) (*github.User, error)
```
Comment (midnightconman) [inline prow/prstatus/prstatus.go]: Can we delete this comment also?

File: prow/prstatus/prstatus.go (hunk 3 above)
Comment (hpandeycodeit) [reply prow/prstatus/prstatus.go]: done

File: prow/prstatus/prstatus.go (hunk 4 above)
Comment (hpandeycodeit) [reply prow/prstatus/prstatus.go]: this too

File: prow/prstatus/prstatus.go (hunk 6 above)
Comment (hpandeycodeit) [reply prow/prstatus/prstatus.go]: done

**Comment by stevekuznetsov** [comment]
This is looking good! If you can squash down your changes to one commit, I think we can merge it :)

**Comment by hpandeycodeit** [comment]
@stevekuznetsov Squashed the commits. let me know if it looks good :) Thanks :)

**Comment by stevekuznetsov** [comment]
Eek, looks like you may have mistakenly brought some other `git` history into your branch. Could you try removing that with:

```sh
$ git pull origin master
$ git checkout github_client
$ git reset --hard master
$ git cherry-pick f2af0a84cfd3523e086767ec1fc248c27a0fd0ec
$ git push
```

**Comment by hpandeycodeit** [comment]
@stevekuznetsov hopefully it should be okay now.

**Comment by stevekuznetsov** [comment]
/lgtm
/approve
/hold
@midnightconman do you want a final look?

**Comment by midnightconman** [comment]
Thank you for the PR! @hpandeycodeit 

/lgtm

**Comment by hpandeycodeit** [comment]
Thank you @stevekuznetsov @midnightconman!! :)

**Pull request merged**